stablehlo/dialect/StablehloOps.td

```diff
@@ -1567,7 +1568,9 @@ def StableHLO_TupleOp : StableHLO_Op<"tuple", [Pure,
   }];
 }
 
-def StableHLO_CompareOp: StableHLO_Op<"compare", [Pure, Elementwise,
+def StableHLO_CompareOp: StableHLO_Op<"compare",
+    [HLO_SpeculatableIfAllInputsStatic, NoMemoryEffect,
+    Elementwise,
```

Comment:
for my clarification...  `Elementwise` Trait is unrelated to speculatability?

Reply:
It's sorta related: https://mlir.llvm.org/doxygen/structmlir_1_1OpTrait_1_1Elementwise.html

The doc says "The shape may be dynamic in which case the op's behaviour is undefined for non-matching shapes.", but it doesn't provide an implementation of `getSpeculatability`.

Reply:
got it, thank you.